src/main/java/ml/shifu/shifu/udf/AbstractTrainerUDF.java

```diff
@@ -42,6 +44,9 @@
  */
 public abstract class AbstractTrainerUDF<T> extends EvalFunc<T> {
 
+    // This cache aims to decrease the memory usage when column config is very large.
+    private static Map<String, List<ColumnConfig>> columnConfigCache = new HashMap<>(1);
```

Comment:
I put the cache in `AbstractTrainerUDF` instead of `CommonUtils`. I think `Utils` class holds a cache is not the best practice. If we need take the action of `load config`, `save config` and `load config`, it will fail due to the cache in `Utils`.